### DISCUSSION TITLE
NetBeans 23+ no longer supports String Templates (JEP 430)

### DISCUSSION THREAD

**Discussion by mbien** [apache/netbeans#7620] [Tips, Tricks and Workarounds]
NetBeans is using the javac API (in form of the nb-javac module) for the Java editor support. Due to the fact that JDK 23 removed the String Templates preview feature ([JEP 430](https://openjdk.org/jeps/430)), NetBeans 23 and later, did also lose the ability to recognize String Templates (even when your project is using JDK 21 still).

workaround:

If you want to continue experimenting with the removed preview of JEP 430, consider keeping a copy of NetBeans [22](https://netbeans.apache.org/download/nb22/), which is the last version with support for it - at least until the next iteration of the feature appears in the JDK again.

references:
 - https://mail.openjdk.org/pipermail/amber-spec-experts/2024-April/004106.html
 - https://github.com/openjdk/jdk/pull/18688
 - https://github.com/apache/netbeans/pull/7484

**Comment by jjazzboss** [comment]
ok it's clear thanks
